lib/driver.js

```diff
@@ -892,7 +892,7 @@ class XCUITestDriver extends BaseDriver {
             try {
               this.cachedWdaStatus =
                 this.cachedWdaStatus || (await this.proxyCommand('/status', 'GET'));
-              await this.startWdaSession(this.opts.bundleId, this.opts.processArguments);
+              await this.startWdaSession(this.opts.bundleId, _.cloneDeep(this.opts.processArguments));
```

Comment:
I think `startWdaSession` method side should take care of that instead of the caller. Something like:

```
--- a/lib/driver.js
+++ b/lib/driver.js
@@ -1343,7 +1343,7 @@ class XCUITestDriver extends BaseDriver {
   }
 
   async startWdaSession(bundleId, processArguments) {
-    const args = processArguments ? processArguments.args || [] : [];
+    const args = processArguments ? _.cloneDeep(processArguments.args) || [] : [];
     if (!_.isArray(args)) {
       throw new Error(
         `processArguments.args capability is expected to be an array. ` +
diff --git a/test/unit/language-specs.js b/test/unit/language-specs.js
index 9014b4d5..7a0e98a8 100644
--- a/test/unit/language-specs.js
+++ b/test/unit/language-specs.js
@@ -121,6 +121,7 @@ describe('language and locale', function () {
       proxySpy.calledOnce.should.be.true;
       proxySpy.firstCall.args[0].should.eql('/session');
       proxySpy.firstCall.args[1].should.eql('POST');
+      desiredCapabilities.processArguments.should.equal(processArguments);
       /** @type {any} */ (proxySpy.firstCall.args[2]).should.eql(expectedWDACapabilities);
     });
   });
```

then, test case can take care of it as well

Reply:
thank you! Updated based on your comment